spot_driver/launch/spot_driver.launch.py

```diff
@@ -30,23 +53,41 @@ def launch_robot_state_publisher(context: LaunchContext, has_arm: LaunchConfigur
     )
     ld.add_action(robot_state_publisher)
 
+    rviz_config_file = PathJoinSubstitution([FindPackageShare("spot_driver"), "rviz", "viz_spot.rviz"])
 
-def generate_launch_description() -> launch.LaunchDescription:
-    config_file = LaunchConfiguration("config_file", default="")
-    config_file_arg = DeclareLaunchArgument(
-        "config_file", description="Path to configuration file for the driver.", default_value=""
+    rviz = launch_ros.actions.Node(
+        package="rviz2",
+        executable="rviz2",
+        name="rviz2",
+        arguments=["-d", rviz_config_file, "-f", parent.perform(context)],
+        output="screen",
+        condition=IfCondition(launch_rviz),
     )
 
-    has_arm = LaunchConfiguration("has_arm")
-    has_arm_arg = DeclareLaunchArgument("has_arm", description="Whether spot has arm", default_value="False")
+    ld.add_action(rviz)
 
-    ld = launch.LaunchDescription([config_file_arg, has_arm_arg])
 
-    spot_driver_node = launch_ros.actions.Node(
-        package="spot_driver", executable="spot_ros2", name="spot_ros2", output="screen", parameters=[config_file]
+def generate_launch_description() -> launch.LaunchDescription:
+    launch_args = []
+
+    launch_args.append(
+        DeclareLaunchArgument("config_file", default_value="", description="Path to configuration file for the driver.")
     )
-    ld.add_action(spot_driver_node)
 
-    ld.add_action(OpaqueFunction(function=launch_robot_state_publisher, args=[has_arm, ld]))
+    launch_args.append(DeclareLaunchArgument("has_arm", default_value="False", description="Whether spot has arm"))
+
+    launch_args.append(
+        DeclareLaunchArgument(
+            "tf_prefix", default_value='""', description="apply namespace prefix to robot links and joints"
```

Comment:
Launching `spot_ros2`  appears to be fine, but adding the `prefix:=<name>` based on the initial comments doesn't appear to change the ROS2 topic names. It also doesn't look like it's a launch argument?

The `tf_prefix` only changes the names of the segments. Is this also meant to change the topic names, like how launching with a namespace did?

Reply:
It does not apply the prefix to the topic name, only to the joints and links defined in the URDF. Would you prefer that they apply the namespace to the topic name as well? or have both a namespace for the topic and prefixes for each frame?

Reply:
I would double check this as a last step with @amessing-bdai and @jbarry-bdai to make sure the prefix behaviour is correct. Otherwise, Rviz and launching the driver work on my end